src/utilities/idf/WorkspaceObject.cpp

```diff
@@ -370,7 +370,7 @@ namespace detail {
       }
 
       // check collection NameConflict
-      if (!uniquelyIdentifiableByName()) {
+      if (!newName.empty() && iddObject().isRequiredField(*index) && !uniquelyIdentifiableByName()) {
```

Comment:
This change is mostly to support OutputMeter having an empty name on construction.  If we change to OutputMeter `Meter Name` we can revert this line.

Reply:
This change created https://github.com/NREL/OpenStudio/issues/4247

`OS:Coil:Heating:Electric` does not have `\required-field` on its Name field, and now unicity of name isn't enforced anymore.